openmeter/billing/worker/autoadvance/autoadvance.go

```diff
@@ -0,0 +1,115 @@
+package billingworkerautoadvance
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"sync"
+	"time"
+
+	"github.com/samber/lo"
+
+	"github.com/openmeterio/openmeter/openmeter/billing"
+)
+
+type AutoAdvancer struct {
+	invoice billing.InvoiceService
+
+	logger *slog.Logger
+}
+
+// All runs aut-advance for all eligible invoices
+func (a *AutoAdvancer) All(ctx context.Context, namespaces []string) error {
+	a.logger.InfoContext(ctx, "listing invoices waiting for auto approval")
+
+	invoices, err := a.ListInvoicesToAdvance(ctx, namespaces, nil)
+	if err != nil {
+		return fmt.Errorf("failed to list invoices to advance: %w", err)
+	}
+
+	a.logger.DebugContext(ctx, "found invoices to approve", "count", len(invoices))
+
+	errChan := make(chan error, len(invoices))
+	closeErrChan := sync.OnceFunc(func() {
+		// Drain channel before closing it
+		var stop bool
+		for !stop {
+			select {
+			case _, ok := <-errChan:
+				if !ok {
+					stop = true
+				}
+			default:
+				stop = true
+			}
+		}
+
+		close(errChan)
+	})
+	defer closeErrChan()
+
+	var wg sync.WaitGroup
+	for _, invoice := range invoices {
+		wg.Add(1)
+
```

Comment:
I would consider adding some kind of limit of in-progress jobs as if anybody creates invoices at the same time for multiple customers we might end up having a lot of these parallel executions.